server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -2919,7 +2917,11 @@ public void testSegmentMemoryTrackedInBreaker() throws Exception {
 
         // Deleting a doc causes its memory to be freed from the breaker
         deleteDoc(primary, "_doc", "0");
-        primary.sync(); // need to sync global checkpoint as the soft-deletes retention MergePolicy depends on it.
+        if (primary.indexSettings().isSoftDeleteEnabled()) {
+            // Need to sync global checkpoint and create a new safe commit as the soft-deletes MergePolicy depends on it.
+            primary.flush(new FlushRequest());
+            primary.forceMerge(new ForceMergeRequest().onlyExpungeDeletes(true));
```

Comment:
why is this needed? the force merge seems not necessary maybe do it randomly

Reply:
Yeah, I was wrong -  a flush is enough. I pushed https://github.com/elastic/elasticsearch/pull/30522/commits/9ae627c3e1cb066eb07a880b4329510a46cb0d73.